### DISCUSSION TITLE
ERROR  Identifier '_unref' has already been declared

### DISCUSSION THREAD

**Discussion by richgcook** [nuxt/nuxt#16928] [Questions]
Since upgrading to rc.9 I've been getting this error on npm run generate `ERROR  Identifier '_unref' has already been declared (Note that you need plugins to import files that are not JavaScript)` any thoughts?

**Comment by TheAlexLichter** [comment]
Can you provide a reproduction?

**Reply by richgcook** [reply]
Let me update my question.

I am getting a similar error `Identifier '_unref' has already been declared (Note that you need plugins to import files that are not JavaScript)` when using `Nuxt 3.0.0-rc.9 with Nitro 0.5.1`

Setting an override to `0.4.4` or `0.4.8` doesn't fix this either.

This only happens on npm run generate whereas npm run build works fine.

```
ERROR  Identifier '_unref' has already been declared (Note that you need plugins to import files that are not JavaScript)                                        09:49:32
file: /Users/rich/dev/www/xxxx/web/components/SliderA.vue:8:196
6: import { default as __nuxt_component_5 } from "/Users/rich/dev/www/xxxx/web/components/ProjectTerms.vue";
7: import { ref, computed, onMounted } from 'vue';
8: import { unref as _unref, resolveComponent as _resolveComponent, withCtx as _withCtx, toDisplayString as _toDisplayString, createTextVNode as _createTextVNode, mergeProps as _mergeProps, unref as _unref } from "vue"
                                                                                                                                                                                                       ^
9: import { ssrRenderComponent as _ssrRenderComponent, ssrRenderAttr as _ssrRenderAttr, ssrRenderAttrs as _ssrRenderAttrs, ssrRenderList as _ssrRenderList, ssrInterpolate as _ssrInterpolate } from "vue/server-renderer"

ERROR  Identifier '_unref' has already been declared (Note that you need plugins to import files that are not JavaScript)                                        09:49:32

  at error (node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:1858:30)
  at Module.error (node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:12412:16)
  at Module.tryParse (node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:12789:25)
  at Module.setSource (node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:12694:24)
  at ModuleLoader.addModuleSource (node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:22136:20)
```

**Reply by richgcook** [reply]
What is strange is that I managed to avoid the bug but changing two things in the `SliderA` component it notes in the error.

On both the root element in the component, and on the component within `ProjectTerms`, has `:data-theme-mode="themeModeStore.getMode" ` set. This was working before upgrading fine, and in another component this also works fine but in this component it seems to break the build. I am using pinia for the store.

I swapped this for a computed function, getting the theme mode from the store, rather than using the store getter directly. This seems to fix the build error but as I note in another component I am using the store getter directly.

I am unsure of the correct approach nevertheless.
